File: php/testcode/securecookie_014.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_cookie_options_array
function securecookie014(BenchmarkRequest $req): BenchmarkResponse {
    $token = bin2hex(random_bytes(32));
    setcookie('auth', $token, [
        'expires' => time() + 3600,
        'path' => '/',
        'secure' => true,
        'httponly' => true,
        'samesite' => 'Strict', // vuln-code-snippet safe-line php_cookie_options_array
    ]);
    return BenchmarkResponse::ok('Auth cookie set');
}
// vuln-code-snippet end php_cookie_options_array

File: ruby/testcode/securecookie_014.rb
require_relative 'shared'

module CookieJar
  def self.signed; SignedJar.new; end
  class SignedJar
    def []=(key, val); end
  end
end

# vuln-code-snippet start ruby_securecookie_signed
def set_signed_cookie(req)
  user_id = req.param('user_id').to_i
  jar = CookieJar.signed
  jar[:user_id] = user_id # vuln-code-snippet safe-line ruby_securecookie_signed
  BenchmarkResponse.ok('signed cookie set')
end
# vuln-code-snippet end ruby_securecookie_signed
